### ISSUE TITLE
Fetching FQDN?

### ISSUE THREAD

**Issue by jszwedko** [svartalf/hostname#13]
Thanks for this crate!

I was just wondering what you thought about adding a mechanism to fetch the fully qualified domain name (similar to `hostname --fqdn`).

**Comment by svartalf** [comment]
Yeah, it would be nice to have it, but it needs a bit more investigation first how would it work both for Windows and POSIX-compatible systems.

So far it looks like [`GetComputerNameExW`](https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getcomputernameexw) in combination with `ComputerNamePhysicalDnsFullyQualified` flag can be used for Windows.
As for POSIX, seems like [this approach](https://unix.stackexchange.com/a/572552) would be our solution.

It sounds promising so far, but it would be better to double check everything first.

**Comment by magistau** [comment]
I should note that GNU libc apparently returns FQDN for `gethostname`: https://www.gnu.org/software/libc/manual/html_node/Host-Identification.html. And macOS does that too.
I mean, I stumbled upon this issue while I was searching why a certain Rust utility gives a different hostname from its original bash version which uses `scutil`.

**Comment by pemensik** [comment]
Note: ``hostname -f`` may not work on many Linux systems, where myhostname nss plugin is used in /etc/nsswitch.conf. If you have it installed and enabled before DNS, it may prevent obtaining FQDN. It just stops search domains appending in DNS. Compare outputs of ``getent ahosts -s dns $HOSTNAME | head -1`` and ``getent ahosts -s myhostname $HOSTNAME | head -1``, when you have relative only hostname set (without dots) and have search domain used in /etc/resolv.conf.

**Comment by pemensik** [comment]
Unfortunately there does not seem to be good alternative machanism, which would be reliable. At least not simple way of getting this by normal system API.
